Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py

```diff
@@ -0,0 +1,471 @@
+            'skipped': self._skip,
+            'reason': self._skip_reason
+        }
+        return outputs
+
+    def create_readable_outputs(self, outputs):
+        """ Sets the human readable output in `_readable_output` attribute.
+
+        :param outputs: (dict) the command outputs.
+        """
+        title = self._action.title() + ' User Results ({})'.format(self._brand)
+
+        if not self._skip:
+            headers = ["brand", "instanceName", "success", "active", "id", "username",
+                       "email", "errorCode", "errorMessage", "details"]
+        else:
+            headers = ["brand", "instanceName", "skipped", "reason"]
+
+        readable_output = tableToMarkdown(
+            name=title,
+            t=outputs,
+            headers=headers,
+            removeNull=True
+        )
+
+        return readable_output
+
+
+class IAMUserProfile:
+    """ A User Profile object class for IAM integrations.
+
+    :type _user_profile: ``str``
+    :param _user_profile: The user profile information.
+
+    :type _user_profile_delta: ``str``
+    :param _user_profile_delta: The user profile delta.
+
+    :type _vendor_action_results: ``list``
+    :param _vendor_action_results: A List of data returned from the vendor.
+
+    :return: None
+    :rtype: ``None``
+    """
+
+    INDICATOR_TYPE = 'User Profile'
+
+    def __init__(self, user_profile, user_profile_delta=None):
+        self._user_profile = safe_load_json(user_profile)
+        self._user_profile_delta = safe_load_json(user_profile_delta) if user_profile_delta else {}
+        self._vendor_action_results = []
+
+    def get_attribute(self, item):
+        return self._user_profile.get(item)
+
+    def to_entry(self):
+        """ Generates a XSOAR IAM entry from the data in _vendor_action_results.
+        Note: Currently we are using only the first element of the list, in the future we will support multiple results.
+
+        :return: A XSOAR entry.
+        :rtype: ``dict``
+        """
+
+        outputs = self._vendor_action_results[0].create_outputs()
+        readable_output = self._vendor_action_results[0].create_readable_outputs(outputs)
+
+        entry_context = {
+            'IAM.UserProfile(val.email && val.email == obj.email)': self._user_profile,
+            'IAM.Vendor(val.instanceName && val.instanceName == obj.instanceName && '
+            'val.email && val.email == obj.email)': outputs
+        }
+
+        return_entry = {
+            'ContentsFormat': EntryFormat.JSON,
+            'Contents': outputs,
+            'EntryContext': entry_context
+        }
+
+        if self._vendor_action_results[0].should_return_error():
+            return_entry['Type'] = EntryType.ERROR
+        else:
+            return_entry['Type'] = EntryType.NOTE
+            return_entry['HumanReadable'] = readable_output
+
+        return return_entry
+
+    def return_outputs(self):
+        return_results(self.to_entry())
```

Comment:
why do even have this?

Reply:
@anara123 Once we removed all the classes from CommonServerPython, we needed to remove also the part from `return_results` function.
Since we can import in CommonServerPython, @dantavori and I, thought that the best solution to keep it the same and return it as a proper entry to the war room is with adding this function.

Reply:
Instead, lets make the return_results less coupled.
In return_results check if the object or objects that passed has attribute `get_entry` function. If yes, then call that function. This way any object that implements `get_entry` function will work